src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -805,7 +805,7 @@ internal static CryptographicAttributeObjectCollection MakeAttributeCollection(A
 
         private static CryptographicAttributeObject MakeAttribute(AttributeAsn attribute)
         {
-            Oid type = new Oid(attribute.AttrType);
+            Oid type = new Oid(attribute.AttrType.Value, attribute.AttrType.Value);
```

Comment:
I'm guessing that if a test was failing it was checking that we avoided the expense of looking up the FriendlyName value during the ASN read.  I guess I'd need to know what test(s) failed to say how to go forward, but it is likely one of:

* Change the test
* Add PopulateFriendlyName=false support to the codegen
* Change the field from Oid to string(Oid)

this particular format of the change seems like it is the most costly in terms of allocations, GC, and lookup expense.

Reply:
I'll dig up the details about the test and follow up.

I don't think it's reasonable to add PopulateFriendlyName=false. The Oid class already supports delayed friendly name lookup when constructed as `new Oid(oidString, null) `. Neither DerSequenceReader nor AsnReader currently does that, but I didn't find any reason not to change it.

Reply:
The failing test was `SignerInfo_AddUnsignedAttribute_Adds`. It constructs an attribute using `AsnEncodedData attribute1 = CreateTimestampToken(1);` and then goes on to read it back with `VerifyAttributesAreEqual(cms.SignerInfos[0].UnsignedAttributes[0].Values[0], attribute1);`. The test failed on `Assert.Equal(expected.Oid.FriendlyName, actual.Oid.FriendlyName);` in `VerifyAttributesAreEqual`.

Reply:
Btw, I tried changing `AsnReader` to use the other `Oid` constructor:

```c#
        public Oid ReadObjectIdentifier(Asn1Tag expectedTag, bool skipFriendlyName=false)
        {
            string oidValue = ReadObjectIdentifierAsString(expectedTag, out int bytesRead);
            Oid oid = skipFriendlyName ? new Oid(oidValue, oidValue) : new Oid(oidValue, null);

            // Don't slice until the return object has been created.
            _data = _data.Slice(bytesRead);

            return oid;
        }
```

It still passes all the tests and avoids the friendly name lookups if the friendly name is not accessed.

Reply:
`AsnReader`'s `ReadObjectIdentifier` takes a bool for `skipFriendlyName`.  The purpose was just to avoid the lookup.

I didn't have internalized (maybe never realized) that `new Oid(value, null)` was a valid "don't populate FriendlyName until it's asked for".  I'd be supportive (in a followup change) of making `ReadObjectIdentifier` always use that construction pattern, and dropping the bool.

As far as this change is concerned, the test seems to be wrong.  It's comparing user input to decoded output; if the test said `new Oid("0.0", "blue light special on aisle 3")` it wouldn't pass in either form.  So I'd put this line of code back, and delete the FriendlyName equivalence assert.

Reply:
Restating since our messages crossed:

> Btw, I tried changing AsnReader to use the other Oid constructor

Please don't add any more concept changes to this PR.  It should be in a different PR.

Reply:
Ok, will do that and submit another PR for the `ReadObjectIdentifier`.

Reply:
> Please don't add any more concept changes to this PR. It should be in a different PR.

Yeah, I am trying to close this PR, not add new stuff to it :-)

Reply:
https://github.com/dotnet/corefx/pull/32161